config/config.go

```diff
@@ -75,29 +69,141 @@ var (
 	SystemUser *user.User
 )
 
-// configStruct is used to serialize our configuration back into a file
+// configStruct is the top-level data structure used to serialize and
+// deserialize our configuration from/to a YAML file
 type configStruct struct {
-	// Email is the email address of the authenticated user.
-	Email string `yaml:"email,omitempty"`
 
 	// LastVersionCheck is the last time when we successfully checked for a gsctl update.
 	// It has no "omitempty", to enforce the output. Marshaling failed otherwise.
 	LastVersionCheck time.Time `yaml:"last_version_check"`
 
+	// Updated is the time when the config has last been written.
+	Updated string `yaml:"updated"`
+
+	// Endpoints is a map of endpoints
+	Endpoints map[string]*endpointConfigStruct `yaml:"endpoints"`
+
+	// SelectedEndpoint is the URL of the selected endpoint
+	SelectedEndpoint string `yaml:"selected_endpoint"`
+
+	// Token is the token found for the selected endpoint. Might be empty.
+	// Not marshalled back to the config file, as it is contained in the
+	// endpoint's entry.
+	Token string `yaml:"-"`
+
+	// Email is the user email found for the selected endpoint. Might be empty.
+	// Not marshalled back to the config file, as it is contained in the
+	// endpoint's entry.
+	Email string `yaml:"-"`
+}
+
+// endpointConfigStruct is used to serialize/deserialize endpoint configuration
+// to/from a config file
+type endpointConfigStruct struct {
```

Comment:
Having `Struct` suffix in a struct name is a little bit awkward.

Reply:
Yeah. Changed.